### PR TITLE
Grant write permission of ssh config to all users

### PR DESCRIPTION
**Pull request by nakabonne**
**What this PR does / why we need it**:
Piped has the opportunity to write something to the ssh config. I'm opening to any concerns.

**Which issue(s) this PR fixes**:

Ref https://github.com/pipe-cd/pipe/issues/1879

**Does this PR introduce a user-facing change?**:

```release-note
NONE
```

### CODE REVIEW

File: dockers/piped-base/Dockerfile [hunk 1]
```diff
@@ -19,7 +19,7 @@ RUN \
         bash && \
     update-ca-certificates && \
     mkdir ${PIPED_TOOLS_DIR} && \
-    chmod 774 /etc/ssh/ssh_config && \
+    chmod 777 /etc/ssh/ssh_config && \
```
Comment (khanhtc1202) [inline dockers/piped-base/Dockerfile:22]: do we need the execute permission for this config file 👀

Comment (nakabonne) [reply dockers/piped-base/Dockerfile:22]: For sure, `776` can be enough.

**Comment by khanhtc1202** [comment]
Just to confirm, will this write permission for piped is required to enable piped adds necessary info and pulls manifests from remote repo such as GitHub right? 🤔

**Comment by nakabonne** [comment]
@khanhtc1202 Yes, Piped writes ssh config to a file when starting.
https://github.com/pipe-cd/pipe/blob/e2d2dffe0ce989411b8f47ffad1e589c3b7557fa/pkg/git/ssh_config.go#L72-L80

**Comment by khanhtc1202** [comment]
Nice, no wonder 👍 
/lgtm

**Comment by nghialv** [comment]
😄 . I think this should be considered carefully. Want a better way instead of giving this to all users.
But it is ok to merge this PR now.

**Pull request merged**